llama_hub/imdb_review/scraper.py

```diff
@@ -0,0 +1,159 @@
+    except NoSuchElementException:
+        title = ""
+
+    try:
+        rating = revs.find_element(
+            By.CLASS_NAME, "rating-other-user-rating"
+        ).text.split("/")[0]
+    except NoSuchElementException:
+        rating = ""
+    re.sub("\n", " ", contents)
+    re.sub("\t", " ", contents)
+    contents.replace("//", "")
+    date = revs.find_element(By.CLASS_NAME, "review-date").text
+    contents = clean_text(contents)
+    return date, contents, rating, title
+
+
+def main_scraper(movie_name: str, webdriver_engine: str = "edge"):
+    os.makedirs("movie_reviews", exist_ok=True)
+
+    ia = imdb.Cinemagoer()
+    movies = ia.search_movie(movie_name)
+    movie_name = movies[0].data["title"] + " " + str(movies[0].data["year"])
+
+    assert movie_name != "", "Please check the movie name that you passed"
+    print(
+        f"Scraping movie reviews for {movie_name}. If you think it is not the right one, the best practice is to pass the movie name and year"
+    )
+    movie_id = movies[0].movieID
+    movie_link = f"https://www.imdb.com/title/tt{movie_id}/reviews/?ref_=tt_ql_2"
+    if webdriver_engine == "edge":
+        driver = webdriver.Edge(service=Service(EdgeChromiumDriverManager().install()))
+    elif webdriver_engine == "google":
+        driver = webdriver.Chrome(service=Service(ChromeDriverManager().install()))
+    elif webdriver_engine == "firefox":
+        driver = webdriver.Firefox(service=Service(GeckoDriverManager().install()))
+
+    driver.get(movie_link)
+    driver.maximize_window()
+
+    driver.execute_script("return document.body.scrollHeight")
+
+    while True:
+        driver.execute_script("window.scrollTo(0, document.body.scrollHeight-250);")
+        try:
+            load_button = driver.find_element(By.CLASS_NAME, "ipl-load-more__button")
+            load_button.click()
+            time.sleep(1)
+        except Exception:
+            print("Load more operation complete")
+            break
+
+    driver.execute_script("window.scrollTo(0, 100);")
+
+    rev_containers = driver.find_elements(By.CLASS_NAME, "review-container")
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=4) as executor:
+        results = executor.map(scrape_data, rev_containers)
+    reviews_date = []
+    reviews_comment = []
+    reviews_rating = []
+    reviews_title = []
+    for result in results:
+        date, contents, rating, title = result
+        reviews_date.append(date)
+
+        reviews_comment.append(contents)
+        reviews_rating.append(rating)
+        reviews_title.append(title)
+
+        # driver.quit()
+    df = pd.DataFrame(
+        columns=["review_date", "review_title", "review_comment", "review_rating"]
+    )
+
+    df["review_date"] = reviews_date
+    df["review_title"] = reviews_title
+    df["review_comment"] = reviews_comment
+    df["review_rating"] = reviews_rating
+
+    # print(df)
+    df.to_csv(f"movie_reviews/{movie_name}.csv", index=False)
```

Comment:
The Dataframe/pandas can be removed since it's not being used more right? if you need it, add an optional argument `generate_csv` to generate or not

```py
generate_csv: Optional[bool] = False
```